2-mandatory/4-stocks.js

```diff
@@ -21,53 +22,84 @@ const CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS = [
+        stockPriceAverage.push(number((stockPriceTotal / stockPrice.length).toFixed(2)));
+    }
+return stockPriceAverage;
 }
 
 /*
     We also want to see what the change in price is from the first day to the last day for each stock.
     Implement the below function, which
-        - Takes this CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS array as input (remember, it's an array of arrays)
+        - Takes this CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS array as input (remember, 
+            it's an array of arrays)
         - Returns an array containing the price change over the last 5 days for each stock.
-            For example, the first element of the resulting array should contain Apple’s (aapl) price change for the last 5 days.
+            For example, the first element of the resulting array should contain 
+            Apple’s (aapl) price change for the last 5 days.
             In this example it would be:
                 (Apple's price on the 5th day) - (Apple's price on the 1st day) = 172.99 - 179.19 = -6.2
-    The price change value should be rounded to 2 decimal places, and should be a number (not a string)
+    The price change value should be rounded to 2 decimal places, 
+    and should be a number (not a string)
 */
 function getPriceChanges(closingPricesForAllStocks) {
     // TODO
+    let priceChange = [];
+    let change;
+for(price of closingPricesForAllStocks){
+    change = number((price[price.length - 1]- price[0]).toFixed(2));
+    priceChange.push(change);
+}
+return priceChange;
 }
 
 /*
-    As part of a financial report, we want to see what the highest price was for each stock in the last 5 days.
+    As part of a financial report, we want to see what the highest price was for each stock in the 
+    last 5 days.
     Implement the below function, which
         - Takes 2 parameters:
-            - the CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS array as input (remember, it's an array of arrays)
+            - the CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS array as input 
+            (remember, it's an array of arrays)
             - the STOCKS array
         - Returns an array of strings describing what the highest price was for each stock.
-            For example, the first element of the array should be: "The highest price of AAPL in the last 5 days was 180.33"
+            For example, the first element of the array should be: "The highest price of AAPL in the
+             last 5 days was 180.33"
             The test will check for this exact string.
     The stock ticker should be capitalised.
     The price should be shown with exactly 2 decimal places.
 */
 function highestPriceDescriptions(closingPricesForAllStocks, stocks) {
     // TODO
+    let highestPrice=[];
+for(let i=0; i<closingPricesForAllStocks.length;i++){
+    highestPrice.push('The highest price of ${stocks[i].toUpperCase()} in the last 5 days was ${Math.max(. closingPricesForAllStocks[i]).toFixed(2)}');
+    return highestPrice;
 }
 
```

Comment:
I think you have two open { here but only one }!

Reply:
I also think you need to use ` instead of ' ?

Reply:
I would try separating this out into several lines, its not giving the answer we're after and I think you'll find it easier to see where things are going wrong